Sources/Widgets/Widgets3D/EllipseWidget/behavior.js

```diff
@@ -9,38 +8,26 @@ export default function widgetBehavior(publicAPI, model) {
 
   model.classHierarchy.push('vtkEllipseWidgetProp');
 
-  publicAPI.setBounds = (bounds) => {
-    if (superClass.setBounds) {
-      superClass.setBounds(bounds);
+  publicAPI.setCorners = (point1, point2) => {
+    if (superClass.setCorners) {
+      superClass.setCorners(point1, point2);
     }
 
-    const center = vtkBoundingBox.getCenter(bounds);
-    const scale3 = vtkBoundingBox.computeScale3(bounds);
-
-    model.shapeHandle.setOrigin(center);
-    model.shapeHandle.setScale3(scale3);
-  };
-
-  publicAPI.setBoundsFromRadius = (center, pointOnCircle) => {
-    const radius = vec3.distance(center, pointOnCircle);
-
-    publicAPI.setBounds([
-      center[0] - radius,
-      center[0] + radius,
-      center[1] - radius,
-      center[1] + radius,
-      center[2] - radius,
-      center[2] + radius,
-    ]);
-  };
-
-  publicAPI.setBoundsFromDiameter = (point1, point2) => {
     const center = [
       0.5 * (point1[0] + point2[0]),
       0.5 * (point1[1] + point2[1]),
       0.5 * (point1[2] + point2[2]),
     ];
 
-    publicAPI.setBoundsFromRadius(center, point1);
+    const d = [0, 0, 0];
+    vec3.subtract(d, point2, center);
```

Comment:
rename `d` to `diameter`.

However that looks more like a `radius`, but an incorrectly computed radius . (ellipse/circle is **contained** within the rectangle defined by corner points)

Reply:
will rename to direction

Reply:
As far as whether it's a radius, diameter, corner to corner, or center to corner, that's properly handled already based on the `BehaviorCategory` [here](https://github.com/Kitware/vtk-js/blob/5f7c6f66c51d44c84d7741006328d4a3de2bd7ca/Sources/Widgets/Widgets3D/ShapeWidget/behavior.js#L289-L316) which can call [those functions](https://github.com/Kitware/vtk-js/blob/5f7c6f66c51d44c84d7741006328d4a3de2bd7ca/Sources/Widgets/Widgets3D/ShapeWidget/behavior.js#L230-L255) to properly extract the true corners that define the rectangle which contains the ellipse/circle.

Some parts of that PR still don't work (scaling, orienting the `CircleContextRepresentation` properly), but that part should.